yt/data_objects/tests/test_covering_grid.py

```diff
@@ -134,30 +129,40 @@ def test_arbitrary_grid():
                     2**ref_level * ds.domain_dimensions)
             assert_almost_equal(cg["density"], ag["density"])
 
-output_00080 = "output_00080/info_00080.txt"
-@requires_file(output_00080)
 def test_octree_cg():
-    ds = load(output_00080)
-    cgrid = ds.covering_grid(0, left_edge=ds.domain_left_edge, dims=ds.domain_dimensions)
+    oct_mask_list = [8, 0, 0, 0, 0, 8, 0, 0,
+                     0, 0, 0, 0, 0, 0, 0, 0,
+                     8, 0, 0, 0, 0, 0, 0, 0,
+                     0]
+    octree_mask = np.array(oct_mask_list, dtype=np.uint8)
+    quantities = {}
+    quantities[('gas', 'density')] = np.ones((22, 1), dtype=float)
+    bbox = np.array([[-10., 10.], [-10., 10.], [-10., 10.]])
+    ds = load_octree(octree_mask=octree_mask, data=quantities, bbox=bbox,
+                     over_refine_factor=0, partial_coverage=0)
+    cgrid = ds.covering_grid(0, left_edge=ds.domain_left_edge,
+                             dims=ds.domain_dimensions)
```

Comment:
I'd call move this code into a new function in `yt/testing.py` named `fake_octree_ds`.

It might also be worthwhile to add a few more AMR levels to the fake octree dataset you're creating. Right now there are only 3 levels (the root oct is fully refined and two of the root oct's child octs are refined), maybe add two or three more AMR levels, just to make sure we're not testing with *too* simple of an AMR structure.

The format of `octree_mask` can be a little confusing, there's a very nice explanation for how to encode an octree in this way in the docs for the hyperion code: http://docs.hyperion-rt.org/en/stable/advanced/indepth_oct.html

(we could probably improve on our own documentation by adapting that explanation into our docs)

Reply:
done